src/lib/token.js

```diff
@@ -6,10 +6,4 @@ export const generateToken = (data, secret, expiresInSeconds) => jwt.sign(
     { expiresIn: `${expiresInSeconds} seconds`, algorithm: "HS256" }
 );
 
-export const verifyAndDecodeToken = (token, secret) => {
-    try {
-        return jwt.verify(token, secret, { algorithm: "HS256" });
-    } catch (err) {
-        return null;
-    }
-};
+export const verifyAndDecodeToken = (token, secret) => jwt.verify(token, secret, { algorithm: "HS256" });
```

Comment:
Why did you move the error handling logic out of this method? I recall you returning different errors based on the thrown exception. Couldn't we abstract those `jwt` exceptions into our custom, more semantic, ones? Food for thought.

Reply:
I agree with this comment, I think the error handling error should be inside and outside a simpler null check